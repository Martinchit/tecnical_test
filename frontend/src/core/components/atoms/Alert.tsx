import styled from 'styled-components';
import { Alert as A } from 'react-bootstrap';

export const Alert = styled(A)`
  width: 100%;
  margin: 0;
`;
