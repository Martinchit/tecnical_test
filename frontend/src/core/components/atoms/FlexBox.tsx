import styled from 'styled-components';

export interface FlexBoxProps {
  flexDirection?: string;
  flexWrap?: string;
  justifyContent?: string;
  alignItems?: string;
  width?: string;
  height?: string;
  margin?: string;
  padding?: string;
}

export const FlexBox = styled.div<FlexBoxProps>`
  display: flex;
  flex-direction: ${({ flexDirection }) => flexDirection ?? 'row'};
  flex-wrap: ${({ flexWrap }) => flexWrap ?? 'wrap'};
  justify-content: ${({ justifyContent }) => justifyContent ?? 'center'};
  align-items: ${({ alignItems }) => alignItems ?? 'center'};
  width: ${({ width }) => width ?? '100%'};
  height: ${({ height }) => height ?? '100%'};
  margin: ${({ margin }) => margin ?? 0};
  padding: ${({ padding }) => padding ?? 0};
`;
