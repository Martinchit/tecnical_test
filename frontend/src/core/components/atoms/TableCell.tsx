import styled from 'styled-components';

export const TableCell = styled.td`
  vertical-align: middle !important;
  text-align: center;
`;
