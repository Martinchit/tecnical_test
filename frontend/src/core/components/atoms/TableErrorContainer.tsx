import styled from 'styled-components';

export const TableErrorContainer = styled.td`
  border-top: none !important;
  padding-top: 0 !important;
`;
