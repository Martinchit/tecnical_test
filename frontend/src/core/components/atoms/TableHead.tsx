import styled from 'styled-components';

export const TableHead = styled.th`
  text-align: center;
`;
