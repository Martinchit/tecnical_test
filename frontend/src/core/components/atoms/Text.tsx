import styled from 'styled-components';

export type WeightProps = 'light' | 'normal' | 'bold' | 'extra-bold';

export interface TextProps {
  fontSize?: number;
  fontWeight?: WeightProps;
  fontColor?: string;
}

const getFontSize = (props) => () => {
  const { fontSize } = props;
  return fontSize ? `${fontSize}em` : '1em';
};

const getFontWeight = (props) => () => {
  const { fontWeight } = props;
  switch (fontWeight) {
    case 'light':
      return '200';
    case 'normal':
      return '400';
    case 'bold':
      return '600';
    case 'extra-bold':
      return '800';
    default:
      return '400';
  }
};

const getFontColor = (props) => () => {
  const { fontColor } = props;
  return fontColor || '#000';
};

const setDefaultValue = () => 'margin: 0px;padding: 0px;border:0';

export const Text = styled.p<TextProps>`
  font-size: ${getFontSize};
  font-weight: ${getFontWeight};
  color: ${getFontColor};
  ${setDefaultValue}
`;
