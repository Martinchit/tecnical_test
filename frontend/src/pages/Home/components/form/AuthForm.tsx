import React from 'react';
import { Button, Form } from 'react-bootstrap';
import { AuthFormContainer } from './AuthFormContainer';
import { FlexBox } from '../../../../core/components/atoms/FlexBox';
import { ErrorMessage } from '../../../../core/components/atoms/ErrorMessage';
import { Alert } from '../../../../core/components/atoms/Alert';
import { AuthType } from '../../../../core/lib/utils/authType';

interface Props {
  onEmailChange: (value: string) => void;
  onPasswordChange: (value: string) => void;
  onConfirmPasswordChange: (value: string) => void;
  onFormSubmit: () => void;
  email: string;
  password: string;
  confirmPassword: string;
  authType: string;
  formError: string;
  authError: string;
}

export const AuthForm: React.FC<Props> = ({
  onEmailChange,
  onPasswordChange,
  onConfirmPasswordChange,
  onFormSubmit,
  email,
  password,
  confirmPassword,
  authType,
  formError,
  authError,
}) => (
  <AuthFormContainer>
    <Form>
      <Form.Group>
        <Form.Label>Email</Form.Label>
        <Form.Control
          type="email"
          name="email"
          placeholder="Enter Email"
          onChange={({ target }) => onEmailChange(target.value)}
          value={email}
        />
      </Form.Group>
      <Form.Group>
        <Form.Label>Password</Form.Label>
        <Form.Control
          type="password"
          name="password"
          placeholder="Enter Password"
          onChange={({ target }) => onPasswordChange(target.value)}
          value={password}
        />
      </Form.Group>
      {authType === AuthType[0] ? (
        <Form.Group>
          <Form.Label>Confirm Password</Form.Label>
          <Form.Control
            type="password"
            name="confirmPassword"
            placeholder="Enter Confirm Password"
            onChange={({ target }) => onConfirmPasswordChange(target.value)}
            value={confirmPassword}
          />
        </Form.Group>
      ) : null}
      {formError.length ? (
        <FlexBox>
          <ErrorMessage>{formError}</ErrorMessage>
        </FlexBox>
      ) : null}

      <FlexBox margin="10% 0 0 0">
        <Button
          onClick={onFormSubmit}
          disabled={formError.length > 0 || !email || !password}
          variant="primary"
        >
          Submit
        </Button>
      </FlexBox>
      {authError ? (
        <FlexBox margin="15px 0 5px 0">
          <Alert variant="danger">
            {authType} Error: {authError}
          </Alert>
        </FlexBox>
      ) : null}
    </Form>
  </AuthFormContainer>
);
