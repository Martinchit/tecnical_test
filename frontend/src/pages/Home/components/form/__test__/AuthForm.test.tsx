import React from 'react';
import { shallow } from 'enzyme';
import { Button, Form } from 'react-bootstrap';
import { ErrorMessage } from '../../../../../core/components/atoms/ErrorMessage';
import { Alert } from '../../../../../core/components/atoms/Alert';

import { AuthForm } from '../AuthForm';

describe('AuthForm', () => {
  const props = {
    onEmailChange: jest.fn(),
    onPasswordChange: jest.fn(),
    onConfirmPasswordChange: jest.fn(),
    onFormSubmit: jest.fn(),
    email: 'email',
    password: 'password',
    confirmPassword: 'confirmPassword',
    authType: 'Log In',
    formError: '',
    authError: '',
  };

  it('renders correctly', () => {
    const wrapper = shallow(<AuthForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders no confirmPassword field', () => {
    const wrapper = shallow(<AuthForm {...props} />);
    expect(wrapper.find(Form.Control).length).toBe(2);
  });

  it('triggers onEmailChange', () => {
    const email = 'testEmail';
    const event = { target: { value: email } };
    const wrapper = shallow(<AuthForm {...props} />);
    wrapper.find(Form.Control).at(0).simulate('change', event);
    expect(props.onEmailChange).toHaveBeenCalled();
    expect(props.onEmailChange).toHaveBeenCalledWith(email);
  });

  it('triggers onPasswordChange', () => {
    const password = 'testPassword';
    const event = { target: { value: password } };
    const wrapper = shallow(<AuthForm {...props} />);
    wrapper.find(Form.Control).at(1).simulate('change', event);
    expect(props.onPasswordChange).toHaveBeenCalled();
    expect(props.onPasswordChange).toHaveBeenCalledWith(password);
  });

  it('renders confirmPassword field', () => {
    const p = { ...props, authType: 'Sign Up' };
    const wrapper = shallow(<AuthForm {...p} />);
    expect(wrapper.find(Form.Control).length).toBe(3);
  });

  it('triggers onConfirmPasswordChange when authType', () => {
    const p = { ...props, authType: 'Sign Up' };
    const confirmPassword = 'testPassword';
    const event = { target: { value: confirmPassword } };
    const wrapper = shallow(<AuthForm {...p} />);
    wrapper.find(Form.Control).at(2).simulate('change', event);
    expect(props.onConfirmPasswordChange).toHaveBeenCalled();
    expect(props.onConfirmPasswordChange).toHaveBeenCalledWith(confirmPassword);
  });

  it('renders Alert', () => {
    const p = { ...props, authError: 'TestError' };
    const wrapper = shallow(<AuthForm {...p} />);
    expect(wrapper.find(Alert).length).toBe(1);
    expect(wrapper.find(Alert).text()).toBe(
      `${p.authType} Error: ${p.authError}`
    );
  });

  it('renders ErrorMessage', () => {
    const formError = 'testError';
    const p = { ...props, formError };
    const wrapper = shallow(<AuthForm {...p} />);
    expect(wrapper.find(ErrorMessage).length).toBe(1);
    expect(wrapper.find(ErrorMessage).text()).toBe(formError);
    expect(wrapper.find(Button).props().disabled).toBeTruthy();
  });
});
