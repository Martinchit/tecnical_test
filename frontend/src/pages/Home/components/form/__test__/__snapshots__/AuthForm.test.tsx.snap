// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm renders correctly 1`] = `
<styled.div>
  <Form
    inline={false}
  >
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Email
      </FormLabel>
      <FormControl
        name="email"
        onChange={[Function]}
        placeholder="Enter Email"
        type="email"
        value="email"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        srOnly={false}
      >
        Password
      </FormLabel>
      <FormControl
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
        value="password"
      />
    </FormGroup>
    <styled.div
      margin="10% 0 0 0"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[MockFunction]}
        type="button"
        variant="primary"
      >
        Submit
      </Button>
    </styled.div>
  </Form>
</styled.div>
`;
